src/com/owncloud/android/ui/activity/UploadFilesActivity.java

```diff
@@ -171,9 +189,46 @@ public void onCreate(Bundle savedInstanceState) {
     public static void startUploadActivityForResult(Activity activity, Account account, int requestCode) {
         Intent action = new Intent(activity, UploadFilesActivity.class);
         action.putExtra(EXTRA_ACCOUNT, (account));
+        action.putExtra(REQUEST_CODE_KEY,requestCode);
         activity.startActivityForResult(action, requestCode);
     }
 
+    private void uploadFromCamera(){
+        Intent pictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (pictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(pictureIntent, REQUEST_IMAGE_CAPTURE);
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode,int resultCode,Intent capturedData){
+        if(requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK){
+            Bundle capturedImageData = capturedData.getExtras();
+            Bitmap capturedImage = (Bitmap) capturedImageData.get("data");
+            Uri capturedImageUri = getImageUri(getApplicationContext(),capturedImage);
+            capturedImagePath = getRealPathFromUri(capturedImageUri);
+            new CheckAvailableSpaceTask().execute();
+        }
+        else if(resultCode == RESULT_CANCELED){
+            setResult(RESULT_CANCELED);
+            finish();
+        }
+    }
+
+    private Uri getImageUri(Context context, Bitmap capturedImage){
```

Comment:
For this as well. If you write `/**` just above the name of the method and press enter , it will create some comments including the params of the method so that you can explain them